### PR TITLE
on-demand ec2 instances for cog creation batch jobs

### PR DESCRIPTION
**Pull request by solomon-negusse**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [x] Make sure you are requesting to pull a topic/feature/bugfix branch (right side). Don't request your master!
- [x] Make sure you are making a pull request against the develop branch (left side). Also you should start your branch off our develop.
- [x] Check the commit's or even all commits' message styles matches our requested structure.
- [x] Check your code additions will fail neither code linting checks nor unit test.

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

COG creation batch jobs run on SPOT instances that get terminated before completion

## What is the new behavior?

COG creation batch jobs run on on-demand instances

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: app/models/pydantic/jobs.py [hunk 1]
```diff
@@ -138,9 +139,9 @@ class PixETLJob(Job):
 
 
 class GDALCOGJob(Job):
-    """Use for creating COG files using GDAL Python docker in PixETL queue."""
+    """Use for creating COG files using GDAL Python docker in COG queue."""
 
-    job_queue = PIXETL_JOB_QUEUE
+    job_queue = COGIFY_JOB_QUEUE
```
Comment (jterry64) [inline app/models/pydantic/jobs.py:144]: The terraform and code looks good, but I don't think we want to name this all specifically as "cogify". That's just one use case for this on demand queue, but we were also interested in using it eventually for other big jobs that often have issues on spot (e.g. like the TCL tile caches). 

So maybe just something like "on-demand-computing" or something? I guess we also have a "pixetl", we should label that more generically at some point too.

**Review by jterry64 — CHANGES_REQUESTED** [review]

File: app/models/pydantic/jobs.py (hunk 1 above)
Comment (solomon-negusse) [reply app/models/pydantic/jobs.py:144]: Good call,  the pixetl one had thrown me off into thinking the queue is coupled with the specific compute environment.

**Review by danscales — APPROVED** [review]
Looks great!

**Review by dmannarino — APPROVED** [review]
Awesome!

**Pull request merged**